index.js

```diff
@@ -1,129 +1,171 @@
-                        catch (error) {
-                            debug("exception stacktrace: ");
-                            debug(error);
-                            return callback(new Error('CAS authentication failed.'));
-                        }
-                    });
-                };
-        }
-        this.options = options;
-        this.parsed = new URL(options.base_url+options.validateUrl);
-        this.client = this.parsed.protocol == 'https'?https:http;
-        return this;
-    },
-    /**
-     * When encountered with an error, actions can be 
-     *  'block': write response with errors
-     *  'pass': pass error to next function
-     *  'ignore': just call next without writing req.cas
-     */
-    validate: function(action) {
-        if(!action||['block', 'pass', 'ignore'].indexOf(action)==-1) action=this.options.action;
-        var self = this;
-        return function(req, res, next) {
-            var doExit = function(status, message) {
-                debug('Status: '+status+' '+message);
-                if(action == 'block') {
-                    res.writeHead(status);
-                    return res.end(message);
-                }
-                var error = new Error(message);
-                error.statusCode = status;
-                next(action=='pass'?error:null);
-            }
-            var ticket = req.query&&req.query.ticket;
-            if(!ticket && req.headers.authorization) {
-                var split = req.headers.authorization.split(' ');
-                if(split[0]=='Bearer') ticket = split[1];
-            }
-            if(!ticket) {
-                debug('no ticket found');
-                return doExit(401, 'Unauthorized. No service ticket found or invalid.');
+module.exports = {
+  set: function(options) {
+    if (typeof options === 'string') {
+      options = {
+        base_url: options
+      };
+    }
+
+    if (!options.base_url) {
+      throw new Error('No CAS Base URL provided');
+    }
+
+    if (!options.version || ['1', '2', '3'].indexOf(options.version + '') === -1) {
+      options.version = '3';
+    }
+
+    if (!options.action || ['block', 'pass', 'ignore'].indexOf(options.action) === -1) {
+      options.action = 'block';
+    }
+
+    switch (options.version) {
+      case '1':
+        options.validateUrl = options.validateUrl || '/validate';
+        options.validate = function(body, callback) {
+          var lines = body.split('\n');
+
+          if (lines[0] === 'yes' && lines.length >= 2) {
+            return callback(null, {
+              user: lines[1]
+            });
+          } else if (lines[0] === 'no') {
+            return callback(new Error('CAS authentication failed.'));
+          } else {
+            return callback(new Error('Response from CAS server was bad.'));
+          }
+        };
+        break;
+      case '2':
+        options.validateUrl = options.validateUrl || '/proxyValidate';
+        break;
```

Comment:
Hi, this break shoul not be here.. if version is 2, set validateUrl to '/proxyValidate' and use code in default option.

Reply:
Why can there not be a `break`?